pkg/cgroup/resolve_container.go

```diff
@@ -122,6 +133,21 @@ func getContainerInfo(cGroupID, pid uint64, withCGroupID bool) (*ContainerInfo,
 		return cinfo, nil
 	}
 
+	// if the containerID is not found, check newContainerIDCache and see if there are retries left
+	if retries, ok := newContainerIDCache[containerID]; ok {
```

Comment:
retrying to resolve pod to fix #594

Reply:
@caiocfer
can you try this commit? I also have built an image with all the changes in this PR in `quay.io/sustainable_computing_io/kepler:pr-665-may-3`

Reply:
Just test it, but it didn't seem to work. Out of curiosity I tried merging this branch with the one in PR 635, but that also didn't work.

Reply:
thanks, what is your running environment? I want to see if I can create a similar environment and test it out here

Reply:
Ubuntu, running with minikube and kube-prometheus for the grafana/prometheus charts.